src/components/Gif.js

```diff
@@ -0,0 +1,101 @@
+import React, {
+    Component
+} from "react";
+import {
+    addFav,
+    withdrawFav
+} from "../Containers/FavPage/action"
+import {
+    connect
+} from 'react-redux'
+
+import {Wraper, ImgStyled} from "./Gif.style";
+
+class Gifs extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            buttonText: "Fav",
+            fav: false
```

Comment:
I think that you are replicating Redux store with the state. redux store should be the only source of true. Plus, If you can avoid states in your components would be better. (States only should be used for little things).